static/compiler.js

```diff
@@ -283,65 +361,138 @@ function Compiler(domRoot, origFilters, windowLocalPrefix, onChangeCallback, lan
+                slot.asmCodeMirror.setOption('lineNumbers', false);
+                slot.asmCodeMirror.setOption('gutters', ['address', 'opcodes']);
             } else {
-                asmCodeMirror.setOption('lineNumbers', true);
-                asmCodeMirror.setOption('gutters', ['CodeMirror-linenumbers']);
+                slot.asmCodeMirror.setOption('lineNumbers', true);
+                slot.asmCodeMirror.setOption('gutters', ['CodeMirror-linenumbers']);
             }
         });
+
         if (filters.colouriseAsm) {
+            // colorise the editor
             cppEditor.operation(function () {
                 $.each(numberedLines.source, function (line, ordinal) {
                     cppEditor.addLineClass(parseInt(line),
                         "background", "rainbow-" + (ordinal % NumRainbowColours));
                 });
             });
-            asmCodeMirror.operation(function () {
-                $.each(numberedLines.asm, function (line, ordinal) {
-                    asmCodeMirror.addLineClass(parseInt(line),
-                        "background", "rainbow-" + (ordinal % NumRainbowColours));
+            // colorise the assembly in slot
+                slot.asmCodeMirror.operation(function () {
+                    $.each(numberedLines.asm, function (line, ordinal) {
+                        slot.asmCodeMirror.addLineClass(parseInt(line),
+                            "background", "rainbow-" + (ordinal % NumRainbowColours));
+                    });
                 });
-            });
         }
     }
 
     function fakeAsm(text) {
         return [{text: text, source: null}];
     }
 
-    function onChange() {
+    // slot's parameters callback
+    // TODO : refactor with onEditorChange : those functions could call an updateSlot(slot)
+    function onParamChange(slot) { 
+        console.log("[CALLBACK] onParamChange() on slot "+slot.id);
         if (ignoreChanges) return;  // Ugly hack during startup.
-        if (pendingTimeout) clearTimeout(pendingTimeout);
-        pendingTimeout = setTimeout(function () {
-            var data = {
-                source: cppEditor.getValue(),
-                compiler: currentCompilerId(),
-                options: $('.compiler_options').val(),
-                filters: currentFilters()
-            };
-            setSetting('compiler', data.compiler);
-            setSetting('compilerOptions', data.options);
-            var stringifiedReq = JSON.stringify(data);
-            if (stringifiedReq == lastRequest) return;
-            lastRequest = stringifiedReq;
-            data.timestamp = new Date();
-            $.ajax({
-                type: 'POST',
-                url: '/compile',
-                dataType: 'json',
-                contentType: 'application/json',
-                data: JSON.stringify(data),
-                success: function (result) {
-                    onCompileResponse(data, result);
+                if (slot.pendingTimeoutInSlot) {
```

Comment:
This is a pretty minor thing to mention but the indentation goes mad here. Can you ensure the indentation is consistent please? makes it easier to see the "real" changes as opposed to the indentation change.